src/EditorFeatures/Core/Implementation/Formatting/FormatCommandHandler.FormatDocument.cs

```diff
@@ -15,20 +15,21 @@
 using Microsoft.CodeAnalysis.Text;
 using Microsoft.VisualStudio.Commanding;
 using Microsoft.VisualStudio.Text.Editor.Commanding.Commands;
-using Microsoft.VisualStudio.Utilities;
 using Roslyn.Utilities;
 using VSCommanding = Microsoft.VisualStudio.Commanding;
 
 namespace Microsoft.CodeAnalysis.Editor.Implementation.Formatting
 {
     internal partial class FormatCommandHandler : ForegroundThreadAffinitizedObject
     {
+        private const string s_experimentName = "CleanupOn";
```

Comment:
any reason this is not put in here?
http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Experiments/IExperimentationService.cs,20

Reply:
Because the experiments are loaded asynchronously after solution load, there's no guarantee the experiment controlling the text here will be loaded by the time the user invokes format document. The likelyhood is low, but this is more robust.

---
In reply to: [204847180](https://github.com/dotnet/roslyn/pull/28625#discussion_r204847180) [](ancestors = 204847180)

Reply:
@333fred not sure what you meant? are you worrying about blocking while workspace dll is loading? this method requires Document, so how that dll not loaded when this code executes?

Reply:
Experiments are loaded some time after solution load. There is no guarantee that because there is a Document, the experiments are loaded.

---
In reply to: [204946549](https://github.com/dotnet/roslyn/pull/28625#discussion_r204946549) [](ancestors = 204946549)

Reply:
@333fred are we talking about same thing? I still not sure what you mean. I am talking about why s_experiementName is here rather than put into places we put all other AB testing names. 
WellKnownExperimentNames

also, IExperimentationService is MEF service. it will get loaded when first used. so, I am not sure what you meant by it is not guaranteed. unless MEF is completely broken.